ExampleMod/Common/Systems/ExampleWorldGenHookingSystem.cs

```diff
@@ -0,0 +1,49 @@
+﻿using MonoMod.Cil;
+using System.Linq;
+using Terraria;
+using Terraria.GameContent.Generation;
+using Terraria.IO;
+using Terraria.ModLoader;
+using Terraria.WorldBuilding;
+
+namespace ExampleMod.Common.Systems
+{
+	// This ModSystem will demonstrate how to IL edit and Detour world generation passes
+	// Since world generation passes are anonymous methods (they don't have a name), they can't be edited the standard way (using IL_xx or On_xx)
+	public class ExampleWorldGenHookingSystem : ModSystem
+	{
+		// All of the registration should take place in load
+		// Generation pass hooks are unloaded manually, so no Unload method is needed
+		public override void Load() {
+			// IL editing the pyramids pass
+			GenPass pyramidsPass = WorldGen.VanillaGenPasses.FirstOrDefault(t => t.Name == "Pyramids");// Locate the generation pass
```

Comment:
Final feedback, on thinking about this more. 
- `VanillaGenPasses` should be an `IDictionary<string, GenPass>`
- Examples should just access the value, expecting it to exist, since nothing can (or should) remove vanilla gen passes.